### ISSUE TITLE
Electron: 1.6.2 update (<1.5) Completely breaks Dragon menu/submenu voice control (updated)

### ISSUE THREAD

**Issue by cleidigh** [Microsoft/vscode#24707]
Development Environment:

OS: Windows 7 Professional SP1

Visual Studio 15 Community \ C++ Tools \ SDK
VS Code: 1.11.1 March release
VS Code: 1.12.0 Insiders March 12
VS Code: 1.12.0 March 7 master branch OSS build

Dragon Naturally Speaking DPI 14 (Voice/Dictation/Control)

Node: 6.10.1
NPM:  3.10.8

Description Of Bug:

Menu handling on Windows 7 started to behave differently
somewhere between March 7 and the Insiders March 12 update.

The problem is somewhat subtle since it may pertain
to voice control using Dragon. I believe for generic
software control in terms of menus, Dragon must utilize
the win32 API and messaging to detect the menus
and control them with voice commands.

Prior to the insiders March 12 (or close to that)
the stable VSC and insiders and OSS build March 7
both the top menu bar items as well as submenu items
were detected controllable by Dragon.

Now Dragon can detect and utilize the top level menus
but the submenus are now unable to be activated.
From of the feedback I can get from Dragon it appears
it detects the list of submenus but cannot control them.

I think this may also affect screen readers, but I do not
know for sure as I do not use one.

I am also thinking it probably has to do with the electron
framework/WindowsNativeInterface which I think has been updated or parts of it recently.

This is pretty big problem for accessibility and unfortunately
a big step back from prior operation.

I'm already working to contribute to the VSC so I can
probably help debugging this.  I can do successive builds
to try and figure out where the submenu started to fail after March 7
but having done this before it takes a long time especially by Voice
and maybe there's already an obvious place to start ?

Thanks

**Comment by cleidigh** [comment]
@bpasero 
I think I've pretty much proven this is an electron upstream issue (unfortunately)
I did a series of tests with the electron starter app just changing electron versions.
Starting with electron 1.5 they updated Chrome and in npm could also be
they are using the new Windows UIA framework.
Either way it breaks Dragon and forces menus to be controlled
only with arrows.

I looked at the electron repository but found nothing to specifically address this.
I did happen to see someone from Nuance (makers of Dragon) posting a question
about electron and one of their extensions. I have to follow up on that. nothing was conclusive

This will affect any voice users and Dragon is basically the only game in town. So hopefully there is
a solution. It may be solved by exposing the menu templates so Dragon can see them.
I will try and find out what Dragon can work with or what changed with electron.

Unless I missed something I don't think VSC is it fault or can do anything yet.

**Comment by bpasero** [comment]
I am sorry to hear that and would suggest to report this to Electron if possible for you (https://github.com/atom/electron).

I just verified that using NVDA I am still getting the menu items announced when they have focus. Note that we only officially support NVDA as screen reader on Windows at the moment.

/cc @chrisdias

**Comment by cleidigh** [comment]
@bpasero 
@chrisdias 
I did go ahead and post an issue on the electron project
Unfortunately over the weekend I seem to be seeing significant performance issues with Dragon
and the latest insiders not seen in the last stable version  
it does not seem consistent so I cannot describe it very well but when it happens a single keypress
action can take several seconds to be performed 
This would effectively make VSC completely unusable for Dragon users and really bum me out 
to not be able to work on the project.

this issue sounds more like an interference with the Windows messaging or COM interfaces
utilized by Dragon. I also have a call into nuance but I'm not optimistic I will get to anybody
there could really address this.

Putting aside the menu issue (which we can get around mostly with shortcuts) I have seen several
issue posts on significant performance issues. do you see any relationship with the new electron code?

**Comment by bpasero** [comment]
@cleidigh do you see these issues also with normal Chrome browser?

**Comment by cleidigh** [comment]
@bpasero 
sorry didn't disappear. some system and other issues the past few days..
chrome does have some menu issues that's why suspected the electron
update. it's issues are slightly different it seems to have a menu index problem
you voice select one item it actually selects the above item
I don't think we'll get any help from Nuance unfortunately 

I still have to do more tests but I might avoid problems by not using voice click on the main menus
this might solve the voice recognition stalling I would use shortcuts instead
it's a shame but apparently Nuance is bad at support I have not been able to get a hold of anyone on the development team. no answer from the electron team either.

I was able to finish my second PR for review so that's good..

time to look for some more things to help on

**Comment by bpasero** [comment]
We will change to custom menus on Windows. For now you can set `"window.titleBarStyle": "custom"`.
